knowledge_repo/app/templates/index-base.html

```diff
@@ -11,13 +11,46 @@
   {% endfor %}
 {% endmacro %}
 
+{% macro pagination(max_pages=10, extremes=True, autohide=True) %}
+  {% if feed_params %}
+    {% set start = feed_params['start'] | default(0) %}
+    {% set results = feed_params['results'] | default(10) %}
+    {% set page = 1 if (start == 0) else (start//results + 1) %}
+    {% set page_count = feed_params['page_count'] | default(1) %}
+
+    {% if autohide and page_count > 1 %}
+      {% set page_nums = pagination_pages(current_page=page, page_count=page_count, max_pages=max_pages, extremes=extremes) %}
+
+      <div class='pagination-bar' role="group">
+        <a href="{{ "javascript:;" if (page == 1) else modify_query(start=(page-2)*results) }}" class="pagination-stepper {{ "disabled" if page == 1 }}">
```

Comment:
In general I agree about using javascript in hrefs, but in this case I felt it is worth it, and I'll note there is no javascript code modified by jinja here.

The href is a javascript void function rather than `#` because that way it does nothing rather than sending people to the top of the page, which was the previous behaviour (and annoying).

If the page is anything other than 1, then the link is pure html and simply the current query modified such that start query parameter is offset as appropriate. Yes, this parsing *can* be done in javascript, but I felt it simpler to do in Python.

Do you agree?